### PR TITLE
Run different spark version integration tests

### PR DESCRIPTION
**Pull request by alexr-bq**
### Summary

Run different spark version integration tests

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -59,7 +59,9 @@ jobs:
           path: ./functional-tests/lib/
       - name: Copy functional tests to home directory of client container
         run: docker exec docker_client_1 cp -r /spark-connector/functional-tests /home
-      - name: Run the integration tests
-        run: docker exec -w /home/functional-tests docker_client_1 sbt run
+      - name: Run the integration tests on Spark 3.0
```
Comment (jonathanl-bq) [inline .github/workflows/main.yml:62]: We're not going to run them in parallel?

Comment (alexr-bq) [reply .github/workflows/main.yml:62]: This was just to test initially. I could set them to run in parallel, my only concern is could we run into any resource limits?

Comment (jonathanl-bq) [reply .github/workflows/main.yml:62]: I don't see why we would. Wouldn't running in sequence take longer and be more costly, if anything?

Comment (alexr-bq) [reply .github/workflows/main.yml:62]: Maybe, I don't know enough about it, my logic was assuming each independent run uses a VM or something, there might be a limit on those

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -60,8 +60,8 @@ jobs:
       - name: Copy functional tests to home directory of client container
         run: docker exec docker_client_1 cp -r /spark-connector/functional-tests /home
       - name: Run the integration tests on Spark 3.0
-        run: docker exec -w /home/functional-tests docker_client_1 sbt run -DsparkVersion="3.0.2"
+        run: docker exec -w /home/functional-tests docker_client_1 sbt run -DsparkVersion="3.0.2" -DhadoopVersion="2.4.0" 
       - name: Run the integration tests on Spark 3.1
-        run: docker exec -w /home/functional-tests docker_client_1 sbt run -DsparkVersion="3.1.1"
+        run: docker exec -w /home/functional-tests docker_client_1 sbt run -DsparkVersion="3.1.1" -DhadoopVersion="3.3.0
```
Comment (ravjotbrar) [inline .github/workflows/main.yml]: Is this the right way to pass env variables to a container?

Comment (ravjotbrar) [reply .github/workflows/main.yml]: Also missing closing double quotes at the end of line 65

Comment (alexr-bq) [reply .github/workflows/main.yml]: it's passing it to the sbt run

**Review by jonathanl-bq — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (alexr-bq) [reply .github/workflows/main.yml:62]: Ok so I think this one parallelization is ok but we shouldn't push it much further if there are more configurations that we need to test.

**Pull request merged**